### ISSUE TITLE
Unable to clear selection

### ISSUE THREAD

**Issue by selvinkuik** [anubra266/choc-autocomplete#68]
codesandbox: https://codesandbox.io/s/choc-ui-packages-codesandbox-forked-76vih?file=/src/App.js

![](https://media4.giphy.com/media/NNkJDe0BGXLTKA5HMM/giphy.gif?cid=790b7611d841edbcb4044a9ecc60df30505565ee7f1c94e3&rid=giphy.gif&ct=g)

Steps to reproduce:

1. Make a selection
2. Clear the text in the input
3. Blur the field

Expected behaviour:

The state of the input is reset so when focus is regained a new selection can be made.

Actual behaviour:

The value of the input is populated in to the field on blur and the input is not cleared.

**Comment by anubra266** [comment]
You're looking for the `freeSolo` prop, and you might want the `selectOnFocus` prop.

**Comment by manoj-savantly** [comment]
@anubra266 No, I guess this is a different use case. 

Once a user has selected a value, what if they decide to not select this field? There is no way to clear the selection right now. 
In `multiple` selection mode, we can do this with the `x` button on the tags, but in single selection mode this is not possible.

Or am I missing something?

**Comment by anubra266** [comment]
Oh I understand what you're saying now @manoj-savantly, Then this is a duplicate of #54, so I'll be closing it in favour of that.

**Comment by f-laino** [comment]
Hi,
I have same issue and using ref (as suggested) doesn't help me.
I have items as "reach" object, e.g.:
```json
[
  {
    "q": "Text one",
    "uri": "text-one",
    "anhor": "hello-by-me"
  },
  {
    "q": "Text two",
    "uri": "text-two",
    "anhor": "hello-by-her"
  }
]
```
where the value is a combination of `uri` and `anchor` (a link), and the behaviour is that in input field appear last selected value, e.g. `/text-one#hello-by-me`, instead empty value or last label (`Text one`)

demo: https://codesandbox.io/s/choc-ui-packages-codesandbox-forked-rfs5o?file=/src/App.js

**Comment by anubra266** [comment]
Hi @f-laino your codesandbox reproduction doesn't show you trying to clear the input. Could you explain better what you're trying to achieve? Thanks.

**Comment by f-laino** [comment]
Expected behaviour is the same of @selvinkuik , so when I search for item that doesn't exists and than focus out from input, the value of input must not change (empty or previous selected item).
This behaviour occurs correctly when items have simple `value` parameter.

Thank you

**Comment by SpencerKaiser** [comment]
@anubra266 just to add onto this, `freeSolo` doesn't solve my issue either. If I select an option and then decide later I want to clear that option, deleting the input field contents and then clicking away just restores the display value of the previously selected item. 

Ideally there would be `restoreOnBlurIfEmpty` prop that defaults to `true` where if the user triggers a blur, `freeSolo` is `false`, and the input has no value, the previously selected value returns. 

However, if `restoreOnBlurIfEmpty` is set to false, triggering a blur has no effect on the value of the input if the inputs current value is empty. Make sense?

Right now I think my only workaround would be to have a slight delay to accommodate blur and then trigger the clearing of the input via the ref you created in #112, but that's a little hacky 😉

**Comment by anubra266** [comment]
@SpencerKaiser the `restoreOnBlurIfEmpty` prop is available in [v4.16.0](https://github.com/anubra266/choc-autocomplete/releases/tag/v4.16.0)

**Comment by SpencerKaiser** [comment]
You're a bad ass 🦾 I'll pull down the new version and check it out!
